extensions/src/main/java/org/pentaho/platform/web/http/api/resources/utils/SystemUtils.java

```diff
@@ -27,76 +25,108 @@
 import org.pentaho.platform.security.policy.rolebased.actions.PublishAction;
 import org.pentaho.platform.security.policy.rolebased.actions.RepositoryCreateAction;
 import org.pentaho.platform.security.policy.rolebased.actions.RepositoryReadAction;
+import org.pentaho.platform.util.StringUtil;
 
 import java.util.Collections;
+import java.util.EnumSet;
 import java.util.List;
 
 public class SystemUtils {
-
-  private static final Log logger = LogFactory.getLog( SystemUtils.class );
-
   public static boolean canAdminister() {
     IAuthorizationPolicy policy = PentahoSystem.get( IAuthorizationPolicy.class );
-    return policy.isAllowed( RepositoryReadAction.NAME ) && policy.isAllowed( RepositoryCreateAction.NAME )
-        && ( policy.isAllowed( AdministerSecurityAction.NAME ) );
+    return policy.isAllowed( RepositoryReadAction.NAME )
+      && policy.isAllowed( RepositoryCreateAction.NAME )
+      && policy.isAllowed( AdministerSecurityAction.NAME );
   }
 
   public static boolean canUpload( String uploadDir ) {
-    IAuthorizationPolicy policy = PentahoSystem.get( IAuthorizationPolicy.class );
+    if ( StringUtil.isEmpty( uploadDir ) ) {
```

Comment:
When `uploadDir` is empty, the function should still evaluate the `isAdmin` part and give an "in principle" answer. The proposed way, the behavior is changing.